From 61f3d7e4e83ba868c5b96ef4af01b8896eb09dec Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Emilio=20Cobos=20=C3=81lvarez?= <emilio@crisal.io>
Date: Wed, 3 Jan 2018 12:01:42 -0600
Subject: [PATCH] servo: Merge #19654 - style: Allow building stylo with rust
 nightly (from emilio:geckolib-nightly); r=jdm

See individual commits for details.

Source-Repo: https://github.com/servo/servo
Source-Revision: f71fb8bddc28060320c233701898541112322d66

--HG--
extra : subtree_source : https%3A//hg.mozilla.org/projects/converted-servo-linear
extra : subtree_revision : 8f8b2ccece94cc9df26c8c41afba04032fe9d506
---
 servo/components/style/gecko/wrapper.rs                  | 2 +-
 servo/components/style/gecko_bindings/sugar/ownership.rs | 1 -
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/servo/components/style/gecko/wrapper.rs b/servo/components/style/gecko/wrapper.rs
index f8a1837de33f0..32871baf50a7d 100644
--- a/servo/components/style/gecko/wrapper.rs
+++ b/servo/components/style/gecko/wrapper.rs
@@ -1787,7 +1787,7 @@ impl<'le> Eq for GeckoElement<'le> {}
 impl<'le> Hash for GeckoElement<'le> {
     #[inline]
     fn hash<H: Hasher>(&self, state: &mut H) {
-        (self.0 as *const _).hash(state);
+        (self.0 as *const RawGeckoElement).hash(state);
     }
 }
 
diff --git a/servo/components/style/gecko_bindings/sugar/ownership.rs b/servo/components/style/gecko_bindings/sugar/ownership.rs
index b0ac00368eaeb..8695efaaa40fe 100644
--- a/servo/components/style/gecko_bindings/sugar/ownership.rs
+++ b/servo/components/style/gecko_bindings/sugar/ownership.rs
@@ -109,7 +109,6 @@ pub unsafe trait HasArcFFI : HasFFI {
     ///
     /// &GeckoType -> &Arc<ServoType>
     fn as_arc<'a>(ptr: &'a &Self::FFIType) -> &'a RawOffsetArc<Self> {
-        debug_assert!(!(ptr as *const _).is_null());
         unsafe {
             transmute::<&&Self::FFIType, &RawOffsetArc<Self>>(ptr)
         }
